stellargraph/core/graph_networkx.py

```diff
@@ -745,21 +748,61 @@ def edges(self, triple=False) -> Iterable[Any]:
     def has_node(self, node: Any) -> bool:
         return self._graph.__contains__(node)
 
-    def neighbors(self, node: Any) -> Iterable[Any]:
+    def _transform_edges(self, edges, get_node, include_edge_weight, edge_types):
+        edge_types_set = set(edge_types) if edge_types is not None else None
+
+        def get(e):
+            if include_edge_weight:
+                return NeighbourWithWeight(
+                    get_node(e), e[2].get(self._edge_weight_label)
+                )
+            return get_node(e)
+
+        def is_correct_type(e):
+            return (
+                edge_types_set is None
+                or e[2].get(self._edge_type_attr) in edge_types_set
+            )
+
+        return {get(e) for e in edges if is_correct_type(e)}
+
+    def _in(self, node, include_edge_weight, edge_types):
+        return self._transform_edges(
+            self._graph.in_edges(node, data=True),
+            lambda e: e[0],
+            include_edge_weight,
+            edge_types,
+        )
+
+    def _out(self, node, include_edge_weight, edge_types):
+        return self._transform_edges(
+            self._graph.out_edges(node, data=True),
+            lambda e: e[1],
+            include_edge_weight,
+            edge_types,
+        )
+
+    def neighbors(
+        self, node: Any, include_edge_weight=False, edge_types=None
+    ) -> Iterable[Any]:
         if self.is_directed():
-            in_nodes = {e[0] for e in self._graph.in_edges(node)}
-            out_nodes = {e[1] for e in self._graph.out_edges(node)}
+            in_nodes = self._in(node, include_edge_weight, edge_types)
+            out_nodes = self._out(node, include_edge_weight, edge_types)
             return in_nodes | out_nodes
         return nx.neighbors(self._graph, node)
```

Comment:
I think this branch in each of these functions needs to handle the new arguments too: if the graph is undirected, it'll go here and not do any of the filtering/transforming:

```python
from stellargraph import StellarGraph
import networkx as nx

def example_weighted_undirected_hin():
     graph = nx.MultiGraph()
     graph.add_nodes_from([0, 1], label="A")
     graph.add_nodes_from([2, 3], label="B")
     graph.add_weighted_edges_from([(0, 1, 0.0), (0, 1, 1.0)], label="AA")
     graph.add_weighted_edges_from([(1, 2, 10.0), (1, 3, 10.0)], label="AB")
     return StellarGraph(graph)

list(example_weighted_undirected_hin().in_nodes(1, include_edge_weight=True))
# [0, 2, 3]
```

Reply:
Thanks, somehow completely looked past that